runtime/runtime-params-estimator/src/estimator_context.rs

```diff
@@ -133,6 +137,48 @@ impl<'c> EstimatorContext<'c> {
             migration_flags: MigrationFlags::default(),
         }
     }
+
+    #[cfg(feature = "protocol_feature_flat_state")]
+    fn create_flat_state_factory(store: Store) -> FlatStateFactory {
+        // function-level `use` statements here to avoid `unused import` warning
+        // when flat state feature is disabled
+        use near_primitives::types::BlockHeight;
+        use near_store::flat_state::{
+            store_helper, BlockInfo, ChainAccessForFlatStorage, FlatStorageState,
+        };
+        use std::collections::HashSet;
+
+        struct ChainAccess;
+
+        impl ChainAccessForFlatStorage for ChainAccess {
+            fn get_block_info(&self, block_hash: &CryptoHash) -> BlockInfo {
+                BlockInfo { hash: block_hash.clone(), prev_hash: Default::default(), height: 0 }
+            }
+
+            fn get_block_hashes_at_height(
+                &self,
+                _block_height: BlockHeight,
+            ) -> HashSet<CryptoHash> {
+                unimplemented!()
```

Comment:
I guess this is okay because we assume the flat head is always at the tip?
Could you please add a comment stating such assumptions, if there are any?

Reply:
yeah, exactly
added explanation as a comment